### ISSUE TITLE
Error C2664 'double __gmpz_get_d_2exp(mpir_si *,mpz_srcptr)': cannot convert argument 1 from 'CGAL::Gmpzf::Exponent *' to 'mpir_si *'

### ISSUE THREAD

**Issue by fh-igd-mueller-roemer** [CGAL/cgal#5909]
## Issue Details

I am getting the following compiler error on 64-bit Windows:
```
Error C2664 'double __gmpz_get_d_2exp(mpir_si *,mpz_srcptr)': cannot convert argument 1 from 'CGAL::Gmpzf::Exponent *' to 'mpir_si *'
```

`Exponent` is unconditionally defined as `long` in `CGAL/GMP/Gmpzf_type.h` but `mpir_si` is `long long` on 64-bit Windows (see documentation https://mpir.org/mpir-2.7.2.pdf Section 3.3 MPIR on Windows x64. Also applies to MPIR 3.0.0 according to the docs). Is it possible to detect MPIR via defines? If not, checking for `_MSC_VER`  (and `_WIN64`) would probably be ok, as the original GMP doesn't work well with Visual Studio.

See also https://github.com/CGAL/cgal/issues/439#issuecomment-268788244

## Source Code

_N/A_

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Windows 64 Bit
* Compiler: Visual Studio 2019 with v14.28 Build Tools
* Release or debug mode: Both
* CGAL version: 5.2.1
* Boost version: 1.76.0
* Other libraries versions if used (Eigen, TBB, etc.): Eigen 3.3.5, MPIR 2.7.2

**Comment by lrineau** [comment]
@mglisse Do you know if any of CGAL is supposed to be compatible with MPIR? As far as I know, we do not test with it.

**Comment by fh-igd-mueller-roemer** [comment]
The goal of https://github.com/CGAL/cgal/pull/1782 was to make it compatible, but it appears not to have been tested on Windows x64

**Comment by mglisse** [comment]
From https://groups.google.com/g/mpir-devel/c/MjJsIuGW224/m/oLiRVpwYAgAJ

> * Dr. William Hart's GitHub repository [1] will move to a continuous release model, whereby it is maintained in a continuous state of usability on Linux and OSX, as tested by Travis/Appveyor CI.
> 
> * Brian Gladman will distribute MSVC builds of MPIR from his GitHub account [2] and website [3].

So it looks like the latest official version of MPIR is not 2.7.2 but what you can get from https://github.com/wbhart/mpir/or https://github.com/BrianGladman/mpir , and I think those have the change to mpz_get_d_2exp that makes CGAL work.

**Comment by fh-igd-mueller-roemer** [comment]
@mglisse Yes, 2.7.2 isn't the most current, but the most recent realeased/tagged version 3.0.0 contains the same Section 3.3 in its documentation: https://mpir.org/mpir-3.0.0.pdf and Brian Gladman's repo doesn't have any releases at all (and his website no longer exists under the given link).

**Comment by mglisse** [comment]
*"continuous release model"* means that there won't be more numbered releases, whatever is in the master branch counts as a release.
